src/docfx/lib/markdown/ExtractTitle.cs

```diff
@@ -13,32 +12,27 @@ namespace Microsoft.Docs.Build
 {
     internal static class ExtractTitle
     {
-        public static MarkdownPipelineBuilder UseExtractTitle(this MarkdownPipelineBuilder builder, StrongBox<string> result)
+        public static MarkdownPipelineBuilder UseExtractTitle(this MarkdownPipelineBuilder builder)
         {
             return builder.Use(document =>
             {
                 var h1 = document.SkipWhile(block => block is YamlFrontMatterBlock).FirstOrDefault() as HeadingBlock;
 
                 if (h1 != null && h1.Level == 1)
                 {
-                    result.Value = RenderTitle(h1);
+                    Markup.Context.TitleHtml = RenderTitle(h1);
                     document.Remove(h1);
                 }
             });
         }
 
         private static string RenderTitle(HeadingBlock h1)
         {
-            using (var writer = new StringWriter())
-            {
-                var renderer = new HtmlRenderer(writer);
-                var pipeline = new MarkdownPipelineBuilder().Build();
-                pipeline.Setup(renderer);
-                renderer.Render(h1);
-                writer.Flush();
+            var writer = new StringWriter();
```

Comment:
>StringWriter [](start = 29, length = 12)

dispose?

Reply:
StringWriter dispose is an no-op

---
In reply to: [201658646](https://github.com/dotnet/docfx/pull/3043#discussion_r201658646) [](ancestors = 201658646)